homeassistant/components/sftp_storage/config_flow.py

```diff
@@ -0,0 +1,187 @@
+                async with (
+                    connect(
+                        host=user_config.host,
+                        port=user_config.port,
+                        options=await self.hass.async_add_executor_job(
+                            get_client_options, user_config
+                        ),
+                    ) as ssh,
+                    ssh.start_sftp_client() as sftp,
+                ):
+                    await sftp.chdir(user_config.backup_location)
+                    await sftp.listdir()
+
+                LOGGER.debug(
+                    "Will register SFTP Storage agent with user@host %s@%s",
+                    user_config.host,
+                    user_config.username,
+                )
+
+            except OSError as e:
+                LOGGER.exception(e)
+                placeholders["error_message"] = str(e)
+                errors["base"] = "os_error"
+            except SFTPStorageInvalidPrivateKey:
+                errors["base"] = "invalid_key"
+            except PermissionDenied as e:
+                placeholders["error_message"] = str(e)
+                errors["base"] = "permission_denied"
+            except SFTPStorageMissingPasswordOrPkey:
+                errors["base"] = "key_or_password_needed"
+            except SFTPNoSuchFile:
+                errors["base"] = "sftp_no_such_file"
+            except SFTPPermissionDenied:
+                errors["base"] = "sftp_permission_denied"
+            except Exception as e:  # noqa: BLE001
+                LOGGER.exception(e)
+                placeholders["error_message"] = str(e)
+                placeholders["exception"] = type(e).__name__
+                errors["base"] = "unknown"
+            else:
+                self._async_abort_entries_match(
+                    {
+                        CONF_HOST: user_config.host,
+                        CONF_PORT: user_config.port,
+                        CONF_BACKUP_LOCATION: user_config.backup_location,
+                    }
+                )
+
+                return self.async_create_entry(
+                    title=f"{user_config.username}@{user_config.host}",
+                    data=user_input,
+                )
+
+        return self.async_show_form(
+            step_id=step_id,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST): str,
+                    vol.Optional(CONF_PORT, default=22): int,
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Optional(CONF_PASSWORD): TextSelector(
+                        config=TextSelectorConfig(type=TextSelectorType.PASSWORD)
+                    ),
+                    vol.Optional(CONF_PRIVATE_KEY_FILE): FileSelector(
+                        FileSelectorConfig(accept="*")
+                    ),
+                    vol.Required(CONF_BACKUP_LOCATION): str,
+                }
+            ),
```

Comment:
Use `add_suggested_values_to_schema` so the user doesn't have to fill in everything again.
Please make sure that works for the `FileSelector`; if it doesn't, exclude it from the suggested_values.